### PR TITLE
feat(eslint-plugin): [no-input-rename] add option to allow some inputs

### LINKED ISSUE #423: [no-input-rename] Add option for whitelist of allowed input renames
I'd like to enable the `no-input-rename` rule in our component library codebase, but we have a lot of components that alias the input 'class' to the property `className` in order to be ES3 compatible. It essentially looks like:

```ts
@Input('class') className: string;
```

Can there be a whitelist of allowed input renames that could be specified in the rule's options? Perhaps something like:

```ts
'@angular-eslint/no-input-rename': ['error', 'class']

// or, potentially even better:

'@angular-eslint/no-input-rename': ['error', { allowedRenames: ['class'] }]
```

### PR DESCRIPTION
**Pull request by GavinWu1991**
Fix #423

### CODE REVIEW

File: packages/eslint-plugin/src/rules/no-input-rename.ts [hunk 1]
```diff
@@ -10,7 +10,11 @@ import {
   isImportedFrom,
 } from '../utils/utils';
 
-type Options = [];
+type Options = [
+  {
+    allowedRenames: string[];
```
Comment (rafaelss95) [inline packages/eslint-plugin/src/rules/no-input-rename.ts:15]: I just checked some examples in `@typescript-eslint` repo ([**`no-this-alias`**](https://github.com/typescript-eslint/typescript-eslint/blob/1c1b572c3000d72cfe665b7afbada0ec415e7855/packages/eslint-plugin/src/rules/no-this-alias.ts#L10) and [**`explicit-module-boundary-types`**](https://github.com/typescript-eslint/typescript-eslint/blob/1c1b572c3000d72cfe665b7afbada0ec415e7855/packages/eslint-plugin/src/rules/explicit-module-boundary-types.ts#L19)) and saw that they usually call options like this as "allowedNames".

```suggestion
    readonly allowedNames?: readonly string[];
```

What do you think about this change?

File: packages/eslint-plugin/src/rules/no-input-rename.ts [hunk 2]
```diff
@@ -64,13 +68,31 @@ export default createESLintRule<Options, MessageIds>({
...
     },
-    schema: [],
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allowedRenames: {
+            type: 'array',
+            items: {
+              type: 'string',
```
Comment (rafaelss95) [inline packages/eslint-plugin/src/rules/no-input-rename.ts:78]:
```suggestion
              type: 'string',
              uniqueItems: true,
```

File: packages/eslint-plugin/src/rules/no-input-rename.ts [hunk 3]
```diff
@@ -64,13 +68,31 @@ export default createESLintRule<Options, MessageIds>({
...
+            },
+            description: 'A whitelist of allowed input renames',
```
Comment (rafaelss95) [inline packages/eslint-plugin/src/rules/no-input-rename.ts:80]: We generally don't fill this schema option, but if we do I would rephrase it to:

```suggestion
            description: 'A list with allowed input names',
```

File: packages/eslint-plugin/src/rules/no-input-rename.ts [hunk 4]
```diff
@@ -93,7 +115,8 @@ export default createESLintRule<Options, MessageIds>({
           propertyAlias: string,
           propertyName: string,
         ): boolean => {
-          return !!(
+          return (
+            allowedRenames.indexOf(propertyAlias) !== -1 ||
```
Comment (rafaelss95) [inline packages/eslint-plugin/src/rules/no-input-rename.ts:119]:
```suggestion
            allowedRenames.includes(propertyAlias) ||
```

**Review by rafaelss95 — COMMENTED** [review]
Thanks @GavinWu1991 for addressing this! I've left some minor comments, lmk what you think.

File: packages/eslint-plugin/src/rules/no-input-rename.ts (hunk 3 above)
Comment (GavinWu1991) [reply packages/eslint-plugin/src/rules/no-input-rename.ts:80]: Thank you for your review, I found that the description of our rules is directly quoted from codelyzer's page, but there are many rules we have enhanced, so that why I just added this part in the schema. 
I think I can try to add an extra readme document in the `docs` directory for this rule since it already supports additional configuration options. I guess this would make it more intuitive for the user. 😀

File: packages/eslint-plugin/src/rules/no-input-rename.ts (hunk 1 above)
Comment (GavinWu1991) [reply packages/eslint-plugin/src/rules/no-input-rename.ts:15]: Good suggestion, I like it.

**Comment by GavinWu1991** [comment]
All the review findings have been remediated, and the description document created for `no-input-rename` rule.

File: packages/eslint-plugin/docs/rules/no-input-rename.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      "allowedNames": ["name"]
+    }
+  ]
+}
+```
+
+Examples of **incorrect** code:
+
+```ts
+import { Input } from '@angular/core';
```
Comment (rafaelss95) [inline packages/eslint-plugin/docs/rules/no-input-rename.md:40]: I think we can remove these imports, no?

**Review by rafaelss95 — APPROVED** [review]
Just one nit, but LGTM! 🚀

**Review by rafaelss95 — APPROVED** [review]

**Comment by JamesHenry** [comment]
Thanks @GavinWu1991!
